timeless/views.py

```diff
@@ -68,48 +68,115 @@ def render_template(self, context):
 
 class ListView(GenericView):
     """ Example:
-        pls add here example of usage, when it will be
+        class SettingsListView(views.ListView):
+            template_name = "restaurants/tables/list.html"
+
+            def get_query(self):
+                return models.Table.query.all()
     """
-    def get_objects(self):
-        """ Method for fetching list of objects from db"""
-        raise NotImplementedError()
 
-    def get(self, *args, **kwargs):
-        """ Method for fetching object from db"""
+    def get_query(self):
+        """ Method determines query"""
         raise NotImplementedError()
 
+    def get(self):
+        """ Fetch list of objects and pass it to template"""
+        objects_list = self.get_query()
+        return self.render_template({"object_list": objects_list})
 
-class DetailView(GenericView):
+
+class SingleObjectMixin:
+    model = None
+
+    def get_object(self, id=None):
+        """ Method fetch object from given model by id """
+        assert self.model, "Model is not provided"
+        return self.model.query.get(id)
+
+
+class DetailView(SingleObjectMixin, GenericView):
```

Comment:
@vladarefiev As I can see `DetailView` and `CreateView` do almost the same. I believe `DetailView` in its generic implementation should not contain form. It should be a class which just passes a single object to template context.

Reply:
@vryazanov ok, agree